### ISSUE TITLE
mysql create database fails on Debian 10

### ISSUE THREAD

**Issue by pierre-b** [robertdebock/ansible-role-mysql#6]
Hi, in order to install a Zabbix server on a Debian 10, I get stuck at the task `robertdebock.mysql : create databases]` with the following playbook

## Playbook
```yaml
---
- name: Prepare
  hosts: all
  gather_facts: yes
  become: yes

  roles:
    - role: robertdebock.bootstrap
    - role: robertdebock.buildtools
    - role: robertdebock.python_pip
      python_pip_modules: 
        - name: PyMySQL
    - role: robertdebock.container_docs
    - role: robertdebock.mysql
      mysql_databases:
        - name: zabbix
          encoding: utf8
          collation: utf8_bin
      mysql_users:
        - name: zabbix
          password: zabbix
          priv: "zabbix.*:ALL"
    - role: robertdebock.zabbix_repository
    - role: robertdebock.zabbix_server
```

## Output

TASK [robertdebock.mysql : create databases] ***********************************************************************
failed: [51.91.148.106] (item=zabbix) => {"ansible_loop_var": "item", "changed": false, "item": {"collation": "utf8_bin", "encoding": "utf8", "name": "zabbix"}, "msg": "unable to find /root/.my.cnf. Exception message: (2003, \"Can't connect to MySQL server on 'localhost' ([Errno 2] No such file or directory)\")"}

## Environment

- Control node OS: Mac OS
- Control node Ansible version: 2.9.6
- Managed node OS: Debian 10

Any idea please? Did I miss something?

Thanks

**Comment by robertdebock** [comment]
I can't reproduce this issue, here is what I have. Please tell me what is different in your/mine setup:

Vagrantfile:
```
Vagrant.configure("2") do |config|
  config.vm.define "debian10" do |debian|
    debian.vm.box = "debian/buster64"
    debian.vm.synced_folder '.', '/vagrant', disabled: true
    debian.vm.provider :libvirt do |domain|
      domain.memory = 1024
      domain.cpus = 1
    end
  end
end
```

mysql.yml:
```yaml
---
- name: Converge
  hosts: all
  become: yes
  gather_facts: no

  roles:
    - role: robertdebock.bootstrap
    - role: robertdebock.alternatives
      alternatives_list:
        - name: python
          path: /usr/bin/python2
          link: /usr/bin/python
          priority: 1
        - name: python
          path: /usr/bin/python3
          link: /usr/bin/python
          priority: 2
    - role: robertdebock.mysql
      mysql_databases:
        - name: my_db
          encoding: utf8
          collation: utf8_bin
      mysql_users:
        - name: my_user
          password: my_pass
          priv: "my_db.*:ALL"

```

Snipped playbook output:
```
TASK [robertdebock.mysql : install mysql] **************************************
changed: [debian10]

TASK [robertdebock.mysql : configure mysql] ************************************
changed: [debian10] => (item=bind-address)
changed: [debian10] => (item=socket)
changed: [debian10] => (item=pid-file)

TASK [robertdebock.mysql : initialize mysql] ***********************************
skipping: [debian10]

RUNNING HANDLER [robertdebock.mysql : restart mysql server] ********************
changed: [debian10]

TASK [robertdebock.mysql : start and enable mysql] *****************************
ok: [debian10]

TASK [robertdebock.mysql : create databases] ***********************************
changed: [debian10] => (item=my_db)

TASK [robertdebock.mysql : create users] ***************************************
changed: [debian10] => (item=None)
changed: [debian10]
```

(I've introduced that `robertdebock.alternatives` to fix issue with the Ansible `pip` module, it picked the wrong version of python.)

Hope this helps you or that you see some difference so I'm able to reproduce.

**Comment by pierre-b** [comment]
The error I encountered was on a fresh VM in the cloud, maybe vagrant uses a different debian10 image.

I've ended up with a manual installation of the components (mysql etc...) and won't have time try it again at the moment.

Feel free to close the issue, I will reopen it if we use the ansible role in the future without success.

If you managed to run it on vagrant it should be fine now.

Thanks Robert
